core/types/accounts/account.go

```diff
@@ -38,35 +40,68 @@ func NewAccount() Account {
 	return a
 }
 
-func (a *Account) encodingLength(forStorage bool) uint {
-	var structLength uint
+func (a *Account) EncodingLengthForStorage() uint {
+	var structLength uint = 1 // 1 byte for fieldset
 	var nonContract = a.IsEmptyCodeHash() && a.IsEmptyRoot()
-	if forStorage && nonContract && a.Balance.Sign() == 0 && a.Nonce == 0 && a.Incarnation == 0 {
+	if nonContract && a.Balance.Sign() == 0 && a.Nonce == 0 && a.Incarnation == 0 {
 		return 1
 	}
 
-	if !forStorage || !nonContract || a.Balance.Sign() != 0 || a.Nonce != 0 || a.Incarnation != 0 {
-		var balanceBytes int
-		if b128.Cmp(&a.Balance) == 1 && a.Balance.Sign() == 1 {
-			balanceBytes = 0
-		} else {
-			balanceBytes = (a.Balance.BitLen() + 7) / 8
+	if !nonContract || a.Balance.Sign() != 0 || a.Nonce != 0 || a.Incarnation != 0 {
```

Comment:
this can be further simplified by treating all fields on their own, and reflecting their presence in the `fieldSet`. If a field is not present, it has default value